packages/api/src/mapping/PropertyValueMapping.ts

```diff
@@ -71,3 +78,9 @@ export interface PropertyValueClientToWire {
   sensorTimeseries: TimeSeriesProperty<string | number>;
   geotimeSeriesReference: GeotimeSeriesProperty<GeoJSON.Point>;
 }
+export type getWirePropertyValueFromClient<
+  T extends
+    | keyof PropertyValueClientToWire
+    | Record<string, keyof PropertyValueClientToWire>,
+> = T extends keyof PropertyValueClientToWire ? PropertyValueClientToWire[T]
+  : T;
```

Comment:
I'm being a bit silly but I'm not understanding why this type works. Why doesn't it need to be nested to be kind of like `T extends keyof PropertyValueClientToWire ? PropertyValueClientToWire[T]
  : {[K in T]: PropertyValueClientToWire[K]};`

Reply:
mmm actually, this is a good point. We should still remap the inner types of the struct. We mainly use this type for agg results which isn't supported for structs yet, but I added some tests to make sure this works as intended